lib/rubocop/cop/rails/table_name_assignment.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop enforces the absence of explicit table name definitions.
+      #
+      # `self.table_name=` should only be used for very good reasons,
+      # such as not having control over the database, or working
+      # on a legacy project.
+      #
+      # If you need to change how your model's name is translated to
+      # a table name, you may want to look at Inflections:
+      # https://api.rubyonrails.org/classes/ActiveSupport/Inflector/Inflections.html
+      #
+      # If you wish to add a prefix in front of your model, or wish to change
+      # the default prefix, `self.table_name_prefix` might better suit your needs:
+      # https://api.rubyonrails.org/classes/ActiveRecord/ModelSchema.html#method-c-table_name_prefix-3D
+      #
+      # A whitelist for STI base classes is built-in to this cop. Any model named `Base` will be excluded.
+      # For more information: https://api.rubyonrails.org/classes/ActiveRecord/Inheritance.html
+      #
+      #   # bad
```

Comment:
Can you insert `@example` tag?

```suggestion
      # @example
      #
      #   # bad
```